### PR TITLE
Fix stickiness plugin by using frontend URL for cookie path

### PR DESCRIPTION
**Pull request by kippandrew**
This PR fixes a bug in the sticky session plugin. 

Currently distinct sticky session cookies are being set for every path component of the request URI. For example, if a frontend is defined as `example.com/foo/`, and a requests for `example.com/foo/` is made, a sticky session cookie is set for for `example.com/foo/`. If a second request for `example.com/foo/bar/` comes in, a second cookie is set for `example.com/foo/bar/`. The bowser treats these as unique cookies (which they are because the paths are different). Eventually after enough requests to unique paths, the browser may not allow any more cookies to be set, which may cause it lose other valid cookies (such as an authentication cookie).

The fix is to only set the cookie for the path of canonical frontend url.

### CODE REVIEW

**Comment by kippandrew** [comment]
@jspiewak

File: lua/src/lib/plugins/stickiness.lua [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
+    return base64.decode(cookie)
+  else
+    return nil
+  end
+end
+M.set_cookie = function(server, frontend, settings)
+  local name = settings.COOKIE
+  local value = base64.encode(server.address)
+  local path = M.extract_path(frontend)
+  local cookie = tostring(url.escape(name)) .. "=" .. tostring(url.escape(value)) .. "; Path=" .. tostring(path) .. "; HttpOnly"
```
Comment (jspiewak) [inline lua/src/lib/plugins/stickiness.lua]: No `Max-Age` as in the removed code from `main.lua`?

Comment (kippandrew) [inline lua/src/lib/plugins/stickiness.lua]: No. We want these to be session cookies.

**Comment by CBarraford** [comment]
the code looks good, didn't pull down and test it myself.

**Pull request merged**